src/vhip/mlmodel/gene_features.py

```diff
@@ -160,13 +160,58 @@ def __init__(self, gene_file: str) -> None:
             raise Exception(
                 "Genes file is not provided or empty. Please provide a valid gene file."
             )
-
+        self.id = gene_file
+        self.genes: List[Gene] = []
+        self.skipped_genes: List[str] = []
         readout = read_annotated_genes(gene_file)
-        self.genes: List[Gene] = [
-            Gene(
-                gene_seq=str(readout[0][out]),
-                gene_id=str(readout[1][out]),
-                gene_product=str(readout[2][out]),
+
+        for out in range(len(readout[0])):
+            try:
+                self.genes.append(
+                    Gene(
+                        gene_seq=str(readout[0][out]),
+                        gene_id=str(readout[1][out]),
+                        gene_product=str(readout[2][out]),
+                    )
+                )
+            except Exception:
+                self.skipped_genes.append(str(readout[1][out]))
+        print(f"{len(self.skipped_genes)} genes skipped")
+
+    def codon_counts(
+        self, threshold_imprecise: int, threshold_skipped_genes: int
+    ) -> None:
+        """Aggregate the counts for each unique codon and imprecise codons across an entire GeneSet.
+
+        Args:
+            threshold_imprecise (int): Number of imprecise (non-ATGC) codons tolerated in a single gene.
+            threshold_skipped_genes (int): Tolerated number of genes in GeneSet that have more than threshold_imprecise codons.
+        Populates the following class attributes:
+            self.codon_dict (str: int): Counts of each unique codon across all genes in the GeneSet.
+            self.imprecise_codons (int): Total number of imprecise codons found in the GeneSet.
+            self.skipped_imprecise_genes (List(str)): IDs of genes in the GeneSet that have more than threshold_imprecise codons.
+        """
+        self.codon_dict = dict.fromkeys(CODON_LIST, 0)
+        self.imprecise_codons: int = 0
+        self.skipped_imprecise_genes: List[str] = []
+
+        counter = 0
+        for gene in self.genes:
+            counter += 1
+            print(f"Analyzing gene 1 of {len(self.genes)}")
+            gene.calculate_codon_counts()
```

Comment:
print statement is wrong? It should be analyzing gene "counter" instead of 1

Reply:
Thanks! done